sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs

```diff
@@ -871,19 +871,36 @@ protected override async Task ValidateProcessingPreconditions(CancellationToken
         /// </summary>
         ///
         /// <param name="partitionId">The identifier of the partition the checkpoint is for.</param>
-        /// <param name="offset">The offset to associate with the checkpoint, indicating that a processor should begin reading form the next event in the stream.</param>
+        /// <param name="offset">The offset to associate with the checkpoint, indicating that a processor should begin reading from the next event in the stream.</param>
         /// <param name="sequenceNumber">An optional sequence number to associate with the checkpoint, intended as informational metadata.  The <paramref name="offset" /> will be used for positioning when events are read.</param>
         /// <param name="cancellationToken">A <see cref="CancellationToken" /> instance to signal a request to cancel the operation.</param>
         ///
+        [EditorBrowsable(EditorBrowsableState.Never)]
         protected override Task UpdateCheckpointAsync(string partitionId,
                                                       long offset,
                                                       long? sequenceNumber,
+                                                      CancellationToken cancellationToken) => UpdateCheckpointAsync(partitionId, new CheckpointStartingPosition(sequenceNumber, null, offset), cancellationToken);
+
+        /// <summary>
+        ///   Creates or updates a checkpoint for a specific partition, identifying a position in the partition's event stream
+        ///   that an event processor should begin reading from.
+        /// </summary>
+        ///
+        /// <param name="partitionId">The identifier of the partition the checkpoint is for.</param>
+        /// <param name="checkpointStartingPosition">The starting position to associate with the checkpoint, indicating that a processor should begin reading from the next event in the stream.</param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken" /> instance to signal a request to cancel the operation.</param>
+        ///
+        protected override Task UpdateCheckpointAsync(string partitionId,
+                                                      CheckpointStartingPosition checkpointStartingPosition,
                                                       CancellationToken cancellationToken)
         {
             cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
 
             Argument.AssertNotNull(partitionId, nameof(partitionId));
-            Argument.AssertInRange(offset, long.MinValue + 1, long.MaxValue, nameof(offset));
+            if (checkpointStartingPosition.Offset.HasValue)
```

Comment:
Do we need to check the value of the sequence number? Based on my investigation seems like any value is fine.

Reply:
We should do the same validation, since we can't be sure that the data on the blob wasn't messed with.   Since we want to explicitly disallow `long.MinValue` and we need the data to fit into a `long`, exactly what we have here looks good.

Reply:
We should keep this offset check as a fallback, too.   We use it if Sequence Number is missing.